tools/generator/Java.Interop.Tools.Generator.CodeGeneration/CodeGenerator.cs

```diff
@@ -395,8 +395,8 @@ public bool WriteFields (List<Field> fields, string indent, GenBase gen, HashSet
 			bool needsProperty = false;
 			foreach (Field f in fields) {
 				if (gen.ContainsName (f.Name)) {
-					Report.Warning (0, Report.WarningFieldNameCollision, "Skipping {0}.{1}, due to a duplicate field, method or nested type name. {2} (Java type: {3})", gen.FullName, f.Name, gen.HasNestedType (f.Name) ? "(Nested type)" : gen.ContainsProperty (f.Name, false) ? "(Property)" : "(Method)", gen.JavaName);
-					continue;
+					f.HasNameCollision = true;
```

Comment:
Would it be possible to pass a flag to `WriteField` indicating there is a name collision on this field rather than creating a new property on the model?

Reply:
Are you suggestion We overload the WriteFile method with an extra arg? 
I can do that, but I am not sure it is the right thing. We would lose track of it later, if we need that information in a different part we would not have. It seems to me that stamping on the object that we did this modification makes more sense than only has this info locally on the method. What is the harm you are seeing of having this extra info stamped on the object?

Reply:
Not overloading, just adding a new parameter to it.  I would like us to move in the direction of not modifying the model during the code writing step.  It means we can't parallelize code writing and it's all a race condition if you want to use that data later because there's no guarantee we will generate in the same order every time.

That leaves us 2 options:

1) Use a new parameter on `WriteFile` since this info is currently only used here.  If we someday need it elsewhere we can worry about it then.
2) Modify the model in a Fixup before we start code writing, like this https://github.com/xamarin/java.interop/blob/master/tools/generator/CodeGenerator.cs#L176.  In this case I wouldn't add the `HasNameCollision` property and would just modify the `Field.Name` property directly.

Reply:
I will add the extra arg on the method, however I think this is not the ideal direction, since we are just adding more and more logic to generator instead of just producing code.
Like I mentioned before in other topics, in my view generator should just produce code, all the fixups should occurs outside the generator.

Reply:
done :)

Reply:
My opinion is logic in `generator` is okay, after all, it has to go somewhere.  I just don't want to do it in the actual code-writing phase (`GenBase.Generate (..)`).  

Long term I would like to move the code-writing phase into a separate library, which is probably more along the lines of what you are thinking.  So you build up the model and then pass it to that new library.  Unfortunately that's a tricky and risky task because of the existing logic intertwined with code-writing and being able to verify we don't break anything.

At the very least, I can try to not add any new mid-write model-changing that will make breaking them apart even harder.